test/cmd/images.sh

```diff
@@ -20,125 +21,125 @@ USE_IMAGES=${USE_IMAGES:-$defaultimage}
-[ -n "$(oc get imageStreams wildfly --template="{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}")" ]
-oc get istag | grep -q "wildfly"
-oc annotate istag/wildfly:latest foo=bar
-oc get istag/wildfly:latest -o jsonpath={.metadata.annotations.foo} | grep -q "bar"
-oc annotate istag/wildfly:latest foo-
-[ ! "$(oc get istag/wildfly:latest -o jsonpath={.metadata.annotations} | grep -q 'bar')" ]
-oc delete imageStreams ruby
-oc delete imageStreams nodejs
-oc delete imageStreams wildfly
-oc delete imageStreams postgresql
-oc delete imageStreams mongodb
-[ -z "$(oc get imageStreams ruby --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams nodejs --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams postgresql --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams mongodb --template="{{.status.dockerImageRepository}}")" ]
-[ -z "$(oc get imageStreams wildfly --template="{{.status.dockerImageRepository}}")" ]
-tryuntil oc get imagestreamTags mysql:latest
-[ -n "$(oc get imagestreams mysql --template="{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}")" ]
-oc describe istag/mysql:latest
-[ "$(oc describe istag/mysql:latest | grep "Environment:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Created:")" ]
-[ "$(oc describe istag/mysql:latest | grep "Image Name:")" ]
+os::cmd::expect_success "tryuntil 'oc get imagestreamtags wildfly:latest'"
+os::cmd::expect_success_and_text "oc get imageStreams wildfly --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
+os::cmd::expect_success_and_text 'oc get istag' 'wildfly'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo=bar'
+os::cmd::expect_success_and_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations.foo}' 'bar'
+os::cmd::expect_success 'oc annotate istag/wildfly:latest foo-'
+os::cmd::expect_success_and_not_text 'oc get istag/wildfly:latest -o jsonpath={.metadata.annotations}' 'bar'
+os::cmd::expect_success 'oc delete imageStreams ruby'
+os::cmd::expect_success 'oc delete imageStreams nodejs'
+os::cmd::expect_success 'oc delete imageStreams wildfly'
+os::cmd::expect_success 'oc delete imageStreams postgresql'
+os::cmd::expect_success 'oc delete imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams ruby'
+os::cmd::expect_failure 'oc get imageStreams nodejs'
+os::cmd::expect_failure 'oc get imageStreams postgresql'
+os::cmd::expect_failure 'oc get imageStreams mongodb'
+os::cmd::expect_failure 'oc get imageStreams wildfly'
+os::cmd::expect_success "tryuntil 'oc get imagestreamTags mysql:latest'"
+os::cmd::expect_success_and_text "oc get imagestreams mysql --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
```

Comment:
@soltysh 
Template error, looks like we haven't waited long enough? Unclear why, as the `tryuntil` is right above this.
```
FAILURE: oc get imagestreams mysql --template='{{ index .metadata.annotations "openshift.io/image.dockerRepositoryCheck"}}': the command returned the wrong error code
Standard output from the command:
Error executing template: reflect: call of reflect.Value.Type on zero Value
template was:
	{{ index .metadata.annotations "openshift.io/image.dockerRepositoryCheck"}}
raw data was:
	{"kind":"ImageStream","apiVersion":"v1","metadata":{"name":"mysql","namespace":"cmd-images","selfLink":"/oapi/v1/namespaces/cmd-images/imagestreams/mysql","uid":"c035bd89-92c0-11e5-9004-54ee753e2644","resourceVersion":"364","creationTimestamp":"2015-11-24T15:33:46Z"},"spec":{"tags":[{"name":"5.5","from":{"kind":"DockerImage","name":"openshift/mysql-55-centos7:latest"}},{"name":"5.6","from":{"kind":"DockerImage","name":"centos/mysql-56-centos7:latest"}},{"name":"latest","from":{"kind":"ImageStreamTag","name":"5.5"}}]},"status":{"dockerImageRepository":"172.30.254.155:5000/cmd-images/mysql","tags":[{"tag":"5.5","items":[{"created":"2015-11-24T15:33:52Z","dockerImageReference":"docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2","image":"sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2"}]},{"tag":"latest","items":[{"created":"2015-11-24T15:33:52Z","dockerImageReference":"docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2","image":"sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2"}]}]}}
object given to template engine was:
	map[kind:ImageStream apiVersion:v1 metadata:map[selfLink:/oapi/v1/namespaces/cmd-images/imagestreams/mysql uid:c035bd89-92c0-11e5-9004-54ee753e2644 resourceVersion:364 creationTimestamp:2015-11-24T15:33:46Z name:mysql namespace:cmd-images] spec:map[tags:[map[name:5.5 from:map[kind:DockerImage name:openshift/mysql-55-centos7:latest]] map[name:5.6 from:map[kind:DockerImage name:centos/mysql-56-centos7:latest]] map[name:latest from:map[kind:ImageStreamTag name:5.5]]]] status:map[dockerImageRepository:172.30.254.155:5000/cmd-images/mysql tags:[map[tag:5.5 items:[map[created:2015-11-24T15:33:52Z dockerImageReference:docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2 image:sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2]]] map[tag:latest items:[map[created:2015-11-24T15:33:52Z dockerImageReference:docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2 image:sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2]]]]]]
Standard error from the command:
error: error executing template '{{ index .metadata.annotations "openshift.io/image.dockerRepositoryCheck"}}': 'reflect: call of reflect.Value.Type on zero Value'
----data----
map[apiVersion:v1 metadata:map[selfLink:/oapi/v1/namespaces/cmd-images/imagestreams/mysql uid:c035bd89-92c0-11e5-9004-54ee753e2644 resourceVersion:364 creationTimestamp:2015-11-24T15:33:46Z name:mysql namespace:cmd-images] spec:map[tags:[map[name:5.5 from:map[kind:DockerImage name:openshift/mysql-55-centos7:latest]] map[name:5.6 from:map[kind:DockerImage name:centos/mysql-56-centos7:latest]] map[name:latest from:map[kind:ImageStreamTag name:5.5]]]] status:map[dockerImageRepository:172.30.254.155:5000/cmd-images/mysql tags:[map[tag:5.5 items:[map[dockerImageReference:docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2 image:sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2 created:2015-11-24T15:33:52Z]]] map[tag:latest items:[map[created:2015-11-24T15:33:52Z dockerImageReference:docker.io/openshift/mysql-55-centos7@sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2 image:sha256:7e5dc80b90b0149a3e05aab3b18ad1d26c9aabef459056c4393b51b05c44c5b2]]]]] kind:ImageStream]
```